easypy/randutils.py

```diff
@@ -41,6 +42,48 @@ def get_chunks(self, offset, end, block_size_range, shuffle=False):
             total_size -= size
             offset += size
 
+def load_dictionary(filename):
+    full_path = os.path.join(os.path.dirname(__file__), 'dictionary', filename)
+    with open(full_path) as dict_file:
+        dictionary = dict_file.read().splitlines()
+
+    dictionary.sort(key=len)
+    return {
+        length: list(group) for length, group
+        in itertools.groupby(dictionary, key=len)
+    }
+
+
+NOUNS = load_dictionary('nouns.txt')
+VERBS = load_dictionary('verbs.txt')
+ADVERBS = load_dictionary('adverbs.txt')
+ADJECTIVES = load_dictionary('adjectives.txt')
+PARTS_OF_SPEECH = (NOUNS, VERBS, ADVERBS, ADJECTIVES)
+
+
+def random_meaningful_name(max_length=50):
+    name = ''
+    remaining = max_length
+
+    if max_length < 3:
+        return random_string(max_length, charset=string.ascii_lowercase)
+
+    for part_group in PARTS_OF_SPEECH:
+        max_part_len = remaining - 1 if name else max_length
+        part_lengths = [k for k in part_group.keys() if k <= max_part_len]
```

Comment:
Oh, I didn't know `in dict.keys()` is not that efficient. I will need to filter by non empty groups though so I can detect when there are no usable word groups.
```
usable_word_groups = [
    part_group[l] for l in range(1, max_part_len)
    if len(part_group)
]
```